substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -2995,6 +2995,84 @@ pub mod pallet {
 			Ok(Pays::No.into())
 		}
 
+		/// Reap the member from the pool. Effective only if their contribution to the pool falls
+		/// below ED.
+		#[pallet::call_index(26)]
+		// FIXME eagr weight functions
+		// #[pallet::weight(T::WeightInfo::reap_member())]
+		#[pallet::weight(1)]
+		pub fn reap_member(
```

Comment:
You raised a great question of whether it needs to be an extrinsic and I was convinced that we can do better.
See [here](https://github.com/paritytech/polkadot-sdk/issues/5009#issuecomment-2382738660).